agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -35,6 +40,27 @@
  */
 public final class RestrictorFactory {
 
+    private static final Pattern URL_PATTERN = Pattern.compile("(http[s]?://)((.+):(.+)@)?(.+)");
+
+    /* default */ static class URLConnectionResult {
+
+        public final URLConnection urlConnection;
+
+        public final String url;
+
+        // URLConnection isn't queryable for "security headers";
+        // make this available for unit testing
+        public final String basicAuthHeaderValue;
+
+        private URLConnectionResult(URLConnection urlConnection, String url,
+                String basicAuthHeaderValue) {
+            this.urlConnection = Objects.requireNonNull(urlConnection);
```

Comment:
Sorry, we can't use `Objects` since its only available since Jdk 1.7. Jolokia is supposed to support still 1.6 (yes I know, but there are still many 1.6 users outthere). For Jolokia 2.0 we will relax this then to at least Java 7 (or maybe alread 8)